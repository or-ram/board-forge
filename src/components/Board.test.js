import React from 'react';
import Board from './Board';

describe('Board Test', () => {
  test('Shallow render', () => {
    const wrapper = mount(<Board />);

    expect(wrapper).toMatchSnapshot();
  });

  test('should show 2 moveable places', () => {
    const wrapper = mount(<Board />);

    wrapper.find({ col: 2, row: 5 }).simulate('click');

    expect(wrapper.find('.field-moveable').length).toEqual(2);
  });

  test('move', () => {
    const wrapper = mount(<Board />);

    wrapper.find({ col: 2, row: 5 }).simulate('click');
    wrapper.find({ col: 3, row: 4 }).simulate('click');

    expect(wrapper.find({ col: 3, row: 4 }).find('svg').length).toEqual(1);
  });

  test('wont move same color twice', () => {
    const wrapper = mount(<Board />);

    wrapper.find({ col: 2, row: 5 }).simulate('click');
    wrapper.find({ col: 3, row: 4 }).simulate('click');

    wrapper.find({ col: 4, row: 5 }).simulate('click');

    expect(wrapper.find('.field-moveable').length).toEqual(0);
  });

  test('wont jump over your color', () => {
    const wrapper = mount(<Board />);

    wrapper.find({ col: 2, row: 5 }).simulate('click');
    wrapper.find({ col: 3, row: 4 }).simulate('click');

    wrapper.find({ col: 4, row: 5 }).simulate('click');

    wrapper.find({ col: 5, row: 2 }).simulate('click');
    wrapper.find({ col: 4, row: 3 }).simulate('click');

    wrapper.find({ col: 4, row: 5 }).simulate('click');

    expect(wrapper.find('.field-moveable').length).toEqual(1);
  });

  test('jump', () => {
    const wrapper = mount(<Board />);

    wrapper.find({ col: 2, row: 5 }).simulate('click');
    wrapper.find({ col: 3, row: 4 }).simulate('click');

    wrapper.find({ col: 4, row: 5 }).simulate('click');

    wrapper.find({ col: 5, row: 2 }).simulate('click');
    wrapper.find({ col: 4, row: 3 }).simulate('click');

    wrapper.find({ col: 4, row: 5 }).simulate('click');

    wrapper.find({ col: 3, row: 4 }).simulate('click');
    wrapper.find({ col: 5, row: 2 }).simulate('click');

    expect(wrapper.find({ col: 5, row: 2 }).find('svg').length).toEqual(1);
    expect(wrapper.find({ col: 4, row: 3 }).find('svg').length).toEqual(0);
  });

  test('double jump', () => {
    const wrapper = mount(<Board />);

    wrapper.find({ col: 2, row: 5 }).simulate('click');
    wrapper.find({ col: 3, row: 4 }).simulate('click');

    wrapper.find({ col: 4, row: 5 }).simulate('click');

    wrapper.find({ col: 5, row: 2 }).simulate('click');
    wrapper.find({ col: 4, row: 3 }).simulate('click');

    wrapper.find({ col: 4, row: 5 }).simulate('click');

    wrapper.find({ col: 3, row: 4 }).simulate('click');
    wrapper.find({ col: 5, row: 2 }).simulate('click');

    wrapper.find({ col: 2, row: 3 }).simulate('click');
    wrapper.find({ col: 3, row: 2 }).simulate('click');

    wrapper.find({ col: 6, row: 5 }).simulate('click');
    wrapper.find({ col: 7, row: 4 }).simulate('click');

    wrapper.find({ col: 4, row: 1 }).simulate('click');
    wrapper.find({ col: 3, row: 2 }).simulate('click');

    wrapper.find({ col: 5, row: 6 }).simulate('click');
    wrapper.find({ col: 6, row: 5 }).simulate('click');

    wrapper.find({ col: 2, row: 3 }).simulate('click');
    wrapper.find({ col: 1, row: 4 }).simulate('click');

    wrapper.find({ col: 0, row: 5 }).simulate('click');
    wrapper.find({ col: 2, row: 3 }).simulate('click');
    wrapper.find({ col: 4, row: 1 }).simulate('click');

    expect(wrapper.find({ col: 4, row: 1 }).find('svg').length).toEqual(1);
  });

  test('move after jump', () => {
    const wrapper = mount(<Board />);

    wrapper.find({ col: 2, row: 5 }).simulate('click');
    wrapper.find({ col: 3, row: 4 }).simulate('click');

    wrapper.find({ col: 5, row: 2 }).simulate('click');
    wrapper.find({ col: 4, row: 3 }).simulate('click');

    wrapper.find({ col: 3, row: 4 }).simulate('click');
    wrapper.find({ col: 5, row: 2 }).simulate('click');

    wrapper.find({ col: 3, row: 2 }).simulate('click');
    wrapper.find({ col: 2, row: 3 }).simulate('click');

    wrapper.find({ col: 6, row: 5 }).simulate('click');
    wrapper.find({ col: 7, row: 4 }).simulate('click');

    wrapper.find({ col: 4, row: 1 }).simulate('click');
    wrapper.find({ col: 3, row: 2 }).simulate('click');

    wrapper.find({ col: 5, row: 6 }).simulate('click');
    wrapper.find({ col: 6, row: 5 }).simulate('click');

    wrapper.find({ col: 2, row: 3 }).simulate('click');
    wrapper.find({ col: 1, row: 4 }).simulate('click');

    wrapper.find({ col: 0, row: 5 }).simulate('click');
    wrapper.find({ col: 2, row: 3 }).simulate('click');
    wrapper.find({ col: 4, row: 1 }).simulate('click');

    wrapper.find({ col: 5, row: 0 }).simulate('click');
    wrapper.find({ col: 3, row: 2 }).simulate('click');

    expect(wrapper.find('.field-moveable').length).toEqual(0);
  });

  test('crowning', () => {
    const wrapper = mount(<Board />);

    wrapper.find({ col: 2, row: 5 }).simulate('click');
    wrapper.find({ col: 3, row: 4 }).simulate('click');

    wrapper.find({ col: 5, row: 2 }).simulate('click');
    wrapper.find({ col: 4, row: 3 }).simulate('click');

    wrapper.find({ col: 3, row: 4 }).simulate('click');
    wrapper.find({ col: 5, row: 2 }).simulate('click');

    wrapper.find({ col: 3, row: 2 }).simulate('click');
    wrapper.find({ col: 2, row: 3 }).simulate('click');

    wrapper.find({ col: 6, row: 5 }).simulate('click');
    wrapper.find({ col: 7, row: 4 }).simulate('click');

    wrapper.find({ col: 4, row: 1 }).simulate('click');
    wrapper.find({ col: 3, row: 2 }).simulate('click');

    wrapper.find({ col: 5, row: 6 }).simulate('click');
    wrapper.find({ col: 6, row: 5 }).simulate('click');

    wrapper.find({ col: 2, row: 3 }).simulate('click');
    wrapper.find({ col: 1, row: 4 }).simulate('click');

    wrapper.find({ col: 0, row: 5 }).simulate('click');
    wrapper.find({ col: 2, row: 3 }).simulate('click');
    wrapper.find({ col: 4, row: 1 }).simulate('click');

    wrapper.find({ col: 5, row: 0 }).simulate('click');
    wrapper.find({ col: 3, row: 2 }).simulate('click');

    wrapper.find({ col: 5, row: 2 }).simulate('click');
    wrapper.find({ col: 4, row: 1 }).simulate('click');

    wrapper.find({ col: 3, row: 2 }).simulate('click');
    wrapper.find({ col: 2, row: 3 }).simulate('click');

    wrapper.find({ col: 4, row: 1 }).simulate('click');
    wrapper.find({ col: 5, row: 0 }).simulate('click');

    expect(wrapper.find({ crowned: true })).toMatchSnapshot();
    expect(wrapper.find({ crowned: true }).length).toEqual(1);

    wrapper.find({ col: 1, row: 2 }).simulate('click');
    wrapper.find({ col: 0, row: 3 }).simulate('click');

    wrapper.find({ col: 5, row: 0 }).simulate('click');

    expect(wrapper.find('.field-moveable').length).toEqual(3);
  });
});
