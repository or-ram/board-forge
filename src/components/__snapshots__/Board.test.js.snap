// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Test Shallow render 1`] = `
<Board>
  <div
    className="board"
  >
    <Field
      col={0}
      key="0"
      moveable={false}
      onFieldClick={[Function]}
      row={0}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={1}
      key="1"
      moveable={false}
      onFieldClick={[Function]}
      row={0}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      >
        <Dark
          crowned={false}
          height={70}
          width={70}
        >
          <div
            className="stone"
          >
            <svg
              height={70}
              viewBox="0 0 80 80"
              width={70}
            >
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="30"
                stroke="black"
                strokeWidth="20"
              />
            </svg>
          </div>
        </Dark>
      </div>
    </Field>
    <Field
      col={2}
      key="2"
      moveable={false}
      onFieldClick={[Function]}
      row={0}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={3}
      key="3"
      moveable={false}
      onFieldClick={[Function]}
      row={0}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      >
        <Dark
          crowned={false}
          height={70}
          width={70}
        >
          <div
            className="stone"
          >
            <svg
              height={70}
              viewBox="0 0 80 80"
              width={70}
            >
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="30"
                stroke="black"
                strokeWidth="20"
              />
            </svg>
          </div>
        </Dark>
      </div>
    </Field>
    <Field
      col={4}
      key="4"
      moveable={false}
      onFieldClick={[Function]}
      row={0}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={5}
      key="5"
      moveable={false}
      onFieldClick={[Function]}
      row={0}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      >
        <Dark
          crowned={false}
          height={70}
          width={70}
        >
          <div
            className="stone"
          >
            <svg
              height={70}
              viewBox="0 0 80 80"
              width={70}
            >
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="30"
                stroke="black"
                strokeWidth="20"
              />
            </svg>
          </div>
        </Dark>
      </div>
    </Field>
    <Field
      col={6}
      key="6"
      moveable={false}
      onFieldClick={[Function]}
      row={0}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={7}
      key="7"
      moveable={false}
      onFieldClick={[Function]}
      row={0}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      >
        <Dark
          crowned={false}
          height={70}
          width={70}
        >
          <div
            className="stone"
          >
            <svg
              height={70}
              viewBox="0 0 80 80"
              width={70}
            >
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="30"
                stroke="black"
                strokeWidth="20"
              />
            </svg>
          </div>
        </Dark>
      </div>
    </Field>
    <Field
      col={0}
      key="8"
      moveable={false}
      onFieldClick={[Function]}
      row={1}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      >
        <Dark
          crowned={false}
          height={70}
          width={70}
        >
          <div
            className="stone"
          >
            <svg
              height={70}
              viewBox="0 0 80 80"
              width={70}
            >
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="30"
                stroke="black"
                strokeWidth="20"
              />
            </svg>
          </div>
        </Dark>
      </div>
    </Field>
    <Field
      col={1}
      key="9"
      moveable={false}
      onFieldClick={[Function]}
      row={1}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={2}
      key="10"
      moveable={false}
      onFieldClick={[Function]}
      row={1}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      >
        <Dark
          crowned={false}
          height={70}
          width={70}
        >
          <div
            className="stone"
          >
            <svg
              height={70}
              viewBox="0 0 80 80"
              width={70}
            >
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="30"
                stroke="black"
                strokeWidth="20"
              />
            </svg>
          </div>
        </Dark>
      </div>
    </Field>
    <Field
      col={3}
      key="11"
      moveable={false}
      onFieldClick={[Function]}
      row={1}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={4}
      key="12"
      moveable={false}
      onFieldClick={[Function]}
      row={1}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      >
        <Dark
          crowned={false}
          height={70}
          width={70}
        >
          <div
            className="stone"
          >
            <svg
              height={70}
              viewBox="0 0 80 80"
              width={70}
            >
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="30"
                stroke="black"
                strokeWidth="20"
              />
            </svg>
          </div>
        </Dark>
      </div>
    </Field>
    <Field
      col={5}
      key="13"
      moveable={false}
      onFieldClick={[Function]}
      row={1}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={6}
      key="14"
      moveable={false}
      onFieldClick={[Function]}
      row={1}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      >
        <Dark
          crowned={false}
          height={70}
          width={70}
        >
          <div
            className="stone"
          >
            <svg
              height={70}
              viewBox="0 0 80 80"
              width={70}
            >
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="30"
                stroke="black"
                strokeWidth="20"
              />
            </svg>
          </div>
        </Dark>
      </div>
    </Field>
    <Field
      col={7}
      key="15"
      moveable={false}
      onFieldClick={[Function]}
      row={1}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={0}
      key="16"
      moveable={false}
      onFieldClick={[Function]}
      row={2}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={1}
      key="17"
      moveable={false}
      onFieldClick={[Function]}
      row={2}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      >
        <Dark
          crowned={false}
          height={70}
          width={70}
        >
          <div
            className="stone"
          >
            <svg
              height={70}
              viewBox="0 0 80 80"
              width={70}
            >
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="30"
                stroke="black"
                strokeWidth="20"
              />
            </svg>
          </div>
        </Dark>
      </div>
    </Field>
    <Field
      col={2}
      key="18"
      moveable={false}
      onFieldClick={[Function]}
      row={2}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={3}
      key="19"
      moveable={false}
      onFieldClick={[Function]}
      row={2}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      >
        <Dark
          crowned={false}
          height={70}
          width={70}
        >
          <div
            className="stone"
          >
            <svg
              height={70}
              viewBox="0 0 80 80"
              width={70}
            >
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="30"
                stroke="black"
                strokeWidth="20"
              />
            </svg>
          </div>
        </Dark>
      </div>
    </Field>
    <Field
      col={4}
      key="20"
      moveable={false}
      onFieldClick={[Function]}
      row={2}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={5}
      key="21"
      moveable={false}
      onFieldClick={[Function]}
      row={2}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      >
        <Dark
          crowned={false}
          height={70}
          width={70}
        >
          <div
            className="stone"
          >
            <svg
              height={70}
              viewBox="0 0 80 80"
              width={70}
            >
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="30"
                stroke="black"
                strokeWidth="20"
              />
            </svg>
          </div>
        </Dark>
      </div>
    </Field>
    <Field
      col={6}
      key="22"
      moveable={false}
      onFieldClick={[Function]}
      row={2}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={7}
      key="23"
      moveable={false}
      onFieldClick={[Function]}
      row={2}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      >
        <Dark
          crowned={false}
          height={70}
          width={70}
        >
          <div
            className="stone"
          >
            <svg
              height={70}
              viewBox="0 0 80 80"
              width={70}
            >
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="30"
                stroke="black"
                strokeWidth="20"
              />
            </svg>
          </div>
        </Dark>
      </div>
    </Field>
    <Field
      col={0}
      key="24"
      moveable={false}
      onFieldClick={[Function]}
      row={3}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={1}
      key="25"
      moveable={false}
      onFieldClick={[Function]}
      row={3}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={2}
      key="26"
      moveable={false}
      onFieldClick={[Function]}
      row={3}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={3}
      key="27"
      moveable={false}
      onFieldClick={[Function]}
      row={3}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={4}
      key="28"
      moveable={false}
      onFieldClick={[Function]}
      row={3}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={5}
      key="29"
      moveable={false}
      onFieldClick={[Function]}
      row={3}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={6}
      key="30"
      moveable={false}
      onFieldClick={[Function]}
      row={3}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={7}
      key="31"
      moveable={false}
      onFieldClick={[Function]}
      row={3}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={0}
      key="32"
      moveable={false}
      onFieldClick={[Function]}
      row={4}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={1}
      key="33"
      moveable={false}
      onFieldClick={[Function]}
      row={4}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={2}
      key="34"
      moveable={false}
      onFieldClick={[Function]}
      row={4}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={3}
      key="35"
      moveable={false}
      onFieldClick={[Function]}
      row={4}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={4}
      key="36"
      moveable={false}
      onFieldClick={[Function]}
      row={4}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={5}
      key="37"
      moveable={false}
      onFieldClick={[Function]}
      row={4}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={6}
      key="38"
      moveable={false}
      onFieldClick={[Function]}
      row={4}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={7}
      key="39"
      moveable={false}
      onFieldClick={[Function]}
      row={4}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={0}
      key="40"
      moveable={false}
      onFieldClick={[Function]}
      row={5}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      >
        <Light
          crowned={false}
          height={70}
          width={70}
        >
          <div
            className="stone"
          >
            <svg
              height={70}
              viewBox="0 0 80 80"
              width={70}
            >
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="38"
                stroke="black"
                strokeWidth="4"
              />
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="22"
                stroke="black"
                strokeWidth="4"
              />
            </svg>
          </div>
        </Light>
      </div>
    </Field>
    <Field
      col={1}
      key="41"
      moveable={false}
      onFieldClick={[Function]}
      row={5}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={2}
      key="42"
      moveable={false}
      onFieldClick={[Function]}
      row={5}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      >
        <Light
          crowned={false}
          height={70}
          width={70}
        >
          <div
            className="stone"
          >
            <svg
              height={70}
              viewBox="0 0 80 80"
              width={70}
            >
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="38"
                stroke="black"
                strokeWidth="4"
              />
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="22"
                stroke="black"
                strokeWidth="4"
              />
            </svg>
          </div>
        </Light>
      </div>
    </Field>
    <Field
      col={3}
      key="43"
      moveable={false}
      onFieldClick={[Function]}
      row={5}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={4}
      key="44"
      moveable={false}
      onFieldClick={[Function]}
      row={5}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      >
        <Light
          crowned={false}
          height={70}
          width={70}
        >
          <div
            className="stone"
          >
            <svg
              height={70}
              viewBox="0 0 80 80"
              width={70}
            >
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="38"
                stroke="black"
                strokeWidth="4"
              />
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="22"
                stroke="black"
                strokeWidth="4"
              />
            </svg>
          </div>
        </Light>
      </div>
    </Field>
    <Field
      col={5}
      key="45"
      moveable={false}
      onFieldClick={[Function]}
      row={5}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={6}
      key="46"
      moveable={false}
      onFieldClick={[Function]}
      row={5}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      >
        <Light
          crowned={false}
          height={70}
          width={70}
        >
          <div
            className="stone"
          >
            <svg
              height={70}
              viewBox="0 0 80 80"
              width={70}
            >
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="38"
                stroke="black"
                strokeWidth="4"
              />
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="22"
                stroke="black"
                strokeWidth="4"
              />
            </svg>
          </div>
        </Light>
      </div>
    </Field>
    <Field
      col={7}
      key="47"
      moveable={false}
      onFieldClick={[Function]}
      row={5}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={0}
      key="48"
      moveable={false}
      onFieldClick={[Function]}
      row={6}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={1}
      key="49"
      moveable={false}
      onFieldClick={[Function]}
      row={6}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      >
        <Light
          crowned={false}
          height={70}
          width={70}
        >
          <div
            className="stone"
          >
            <svg
              height={70}
              viewBox="0 0 80 80"
              width={70}
            >
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="38"
                stroke="black"
                strokeWidth="4"
              />
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="22"
                stroke="black"
                strokeWidth="4"
              />
            </svg>
          </div>
        </Light>
      </div>
    </Field>
    <Field
      col={2}
      key="50"
      moveable={false}
      onFieldClick={[Function]}
      row={6}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={3}
      key="51"
      moveable={false}
      onFieldClick={[Function]}
      row={6}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      >
        <Light
          crowned={false}
          height={70}
          width={70}
        >
          <div
            className="stone"
          >
            <svg
              height={70}
              viewBox="0 0 80 80"
              width={70}
            >
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="38"
                stroke="black"
                strokeWidth="4"
              />
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="22"
                stroke="black"
                strokeWidth="4"
              />
            </svg>
          </div>
        </Light>
      </div>
    </Field>
    <Field
      col={4}
      key="52"
      moveable={false}
      onFieldClick={[Function]}
      row={6}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={5}
      key="53"
      moveable={false}
      onFieldClick={[Function]}
      row={6}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      >
        <Light
          crowned={false}
          height={70}
          width={70}
        >
          <div
            className="stone"
          >
            <svg
              height={70}
              viewBox="0 0 80 80"
              width={70}
            >
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="38"
                stroke="black"
                strokeWidth="4"
              />
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="22"
                stroke="black"
                strokeWidth="4"
              />
            </svg>
          </div>
        </Light>
      </div>
    </Field>
    <Field
      col={6}
      key="54"
      moveable={false}
      onFieldClick={[Function]}
      row={6}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={7}
      key="55"
      moveable={false}
      onFieldClick={[Function]}
      row={6}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      >
        <Light
          crowned={false}
          height={70}
          width={70}
        >
          <div
            className="stone"
          >
            <svg
              height={70}
              viewBox="0 0 80 80"
              width={70}
            >
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="38"
                stroke="black"
                strokeWidth="4"
              />
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="22"
                stroke="black"
                strokeWidth="4"
              />
            </svg>
          </div>
        </Light>
      </div>
    </Field>
    <Field
      col={0}
      key="56"
      moveable={false}
      onFieldClick={[Function]}
      row={7}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      >
        <Light
          crowned={false}
          height={70}
          width={70}
        >
          <div
            className="stone"
          >
            <svg
              height={70}
              viewBox="0 0 80 80"
              width={70}
            >
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="38"
                stroke="black"
                strokeWidth="4"
              />
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="22"
                stroke="black"
                strokeWidth="4"
              />
            </svg>
          </div>
        </Light>
      </div>
    </Field>
    <Field
      col={1}
      key="57"
      moveable={false}
      onFieldClick={[Function]}
      row={7}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={2}
      key="58"
      moveable={false}
      onFieldClick={[Function]}
      row={7}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      >
        <Light
          crowned={false}
          height={70}
          width={70}
        >
          <div
            className="stone"
          >
            <svg
              height={70}
              viewBox="0 0 80 80"
              width={70}
            >
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="38"
                stroke="black"
                strokeWidth="4"
              />
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="22"
                stroke="black"
                strokeWidth="4"
              />
            </svg>
          </div>
        </Light>
      </div>
    </Field>
    <Field
      col={3}
      key="59"
      moveable={false}
      onFieldClick={[Function]}
      row={7}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={4}
      key="60"
      moveable={false}
      onFieldClick={[Function]}
      row={7}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      >
        <Light
          crowned={false}
          height={70}
          width={70}
        >
          <div
            className="stone"
          >
            <svg
              height={70}
              viewBox="0 0 80 80"
              width={70}
            >
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="38"
                stroke="black"
                strokeWidth="4"
              />
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="22"
                stroke="black"
                strokeWidth="4"
              />
            </svg>
          </div>
        </Light>
      </div>
    </Field>
    <Field
      col={5}
      key="61"
      moveable={false}
      onFieldClick={[Function]}
      row={7}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
    <Field
      col={6}
      key="62"
      moveable={false}
      onFieldClick={[Function]}
      row={7}
    >
      <div
        className="field field-dark"
        onClick={[Function]}
      >
        <Light
          crowned={false}
          height={70}
          width={70}
        >
          <div
            className="stone"
          >
            <svg
              height={70}
              viewBox="0 0 80 80"
              width={70}
            >
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="38"
                stroke="black"
                strokeWidth="4"
              />
              <circle
                cx="40"
                cy="40"
                fill="white"
                r="22"
                stroke="black"
                strokeWidth="4"
              />
            </svg>
          </div>
        </Light>
      </div>
    </Field>
    <Field
      col={7}
      key="63"
      moveable={false}
      onFieldClick={[Function]}
      row={7}
    >
      <div
        className="field field-light"
        onClick={[Function]}
      />
    </Field>
  </div>
</Board>
`;

exports[`Board Test crowning 1`] = `
<Light
  crowned={true}
  height={70}
  width={70}
>
  <div
    className="stone"
  >
    <svg
      height={70}
      viewBox="0 0 80 80"
      width={70}
    >
      <circle
        cx="40"
        cy="40"
        fill="white"
        r="38"
        stroke="black"
        strokeWidth="4"
      />
      <circle
        cx="40"
        cy="40"
        fill="white"
        r="22"
        stroke="black"
        strokeWidth="4"
      />
      <circle
        cx="40"
        cy="40"
        fill="black"
        r="14"
      />
    </svg>
  </div>
</Light>
`;
